File: templates/perl/template.pl
#!/usr/bin/perl
# Pass 1 or 2 as a cli argument to solve for part 1 or part 2

use v5.30;
use strict;
use warnings;
no warnings 'experimental';

my $inputfile = '../../input/0';
open(FH, '<', $inputfile) or die $!;

my @input = ();
while(<FH>){
    push(@input, $_)
}

sub part1() {
    return 1
}

sub part2() {
    return 2
}

if($ARGV[0] == 1){
    print part1(), "\n";
} else {
    print part2(), "\n";
}
close(FH);

File: templates/python/template.py
#!/usr/bin/env python3

import time
startTime = time.time()

def puzzle(input, part2):
    if part2:
        return 42
    return int(5/7)

f = open("2023/input/1", "r")
input = f.read()

print("2023 - Day 1 - Part 1:", puzzle(input, False))
print("2023 - Day 1 - Part 2:", puzzle(input, True))
print("Execution time: %.3fµs" % ((time.time() - startTime)*1000))